<?xml version="1.0" encoding="UTF-8"?>
<testsuite name="tetris.GameOverTests" tests="3" skipped="0" failures="3" errors="0" timestamp="2022-10-04T11:42:20" hostname="DESKTOP-JDSIO1T" time="0.02">
  <properties/>
  <testcase name="testGameOver" classname="tetris.GameOverTests" time="0.006">
    <failure message="org.scalatest.exceptions.TestFailedException: didPass was false&#10;Hint: The game is over if the blocks of a newly spawned tetromino are already occupied&#10;--------------------------------------------------------------------------------&#10;This is what went wrong:&#10;&#10;step=0, rand=2, actions=&lt;&gt;&#10;Want | Got ✓&#10;------------&#10;..L. | ..L. &#10;LLL. | LLL. &#10;.... | .... &#10;.... | .... &#10;&#10;step=1, rand=2, actions=&lt;Down&gt;&#10;Want | Got ✓&#10;------------&#10;.... | .... &#10;..L. | ..L. &#10;LLL. | LLL. &#10;.... | .... &#10;&#10;step=2, rand=2, actions=&lt;Down&gt;&#10;Want | Got ✓&#10;------------&#10;.... | .... &#10;.... | .... &#10;..L. | ..L. &#10;LLL. | LLL. &#10;&#10;step=3, rand=3, actions=&lt;Down&gt;&#10;Want | Got ✓&#10;------------&#10;.OO. | .OO. &#10;.OO. | .OO. &#10;..L. | ..L. &#10;LLL. | LLL. &#10;&#10;step=4, rand=4, actions=&lt;Down&gt;&#10;Want              | Got ✗&#10;-------------------------&#10;GameOverDisplay() | .SS. &#10;                  | SSO. &#10;                  | ..L. &#10;                  | LLL. &#10;&#10;" type="org.scalatest.exceptions.TestFailedException">org.scalatest.exceptions.TestFailedException: didPass was false
Hint: The game is over if the blocks of a newly spawned tetromino are already occupied
--------------------------------------------------------------------------------
This is what went wrong:

step=0, rand=2, actions=&lt;&gt;
Want | Got ✓
------------
..L. | ..L. 
LLL. | LLL. 
.... | .... 
.... | .... 

step=1, rand=2, actions=&lt;Down&gt;
Want | Got ✓
------------
.... | .... 
..L. | ..L. 
LLL. | LLL. 
.... | .... 

step=2, rand=2, actions=&lt;Down&gt;
Want | Got ✓
------------
.... | .... 
.... | .... 
..L. | ..L. 
LLL. | LLL. 

step=3, rand=3, actions=&lt;Down&gt;
Want | Got ✓
------------
.OO. | .OO. 
.OO. | .OO. 
..L. | ..L. 
LLL. | LLL. 

step=4, rand=4, actions=&lt;Down&gt;
Want              | Got ✗
-------------------------
GameOverDisplay() | .SS. 
                  | SSO. 
                  | ..L. 
                  | LLL. 


	at app//org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
	at app//org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:529)
	at app//org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
	at app//org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:503)
	at app//generic.GameTestSuite.checkGame(GameTestSuite.scala:298)
	at app//tetris.TetrisTestSuiteBase.checkGame(TetrisTestSuiteBase.scala:103)
	at app//tetris.GameOverTests.$anonfun$new$1(GameOverTests.scala:27)
	at app//scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at app//infrastructure.TestBase.$anonfun$test$1(TestBase.scala:36)
	at app//org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at app//org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at app//org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at app//org.scalatest.Transformer.apply(Transformer.scala:22)
	at app//org.scalatest.Transformer.apply(Transformer.scala:20)
	at app//org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:186)
	at app//org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at app//org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at app//infrastructure.TestBase.org$scalatest$concurrent$TimeLimitedTests$$super$withFixture(TestBase.scala:9)
	at app//org.scalatest.concurrent.TimeLimitedTests.$anonfun$withFixture$3(TimeLimitedTests.scala:154)
	at app//org.scalatest.enablers.Timed$$anon$1.timeoutAfter(Timed.scala:127)
	at app//org.scalatest.concurrent.TimeLimits.failAfterImpl(TimeLimits.scala:239)
	at app//org.scalatest.concurrent.TimeLimits.failAfterImpl$(TimeLimits.scala:233)
	at app//org.scalatest.concurrent.TimeLimits$.failAfterImpl(TimeLimits.scala:305)
	at app//org.scalatest.concurrent.TimeLimitedTests.withFixture(TimeLimitedTests.scala:153)
	at app//org.scalatest.concurrent.TimeLimitedTests.withFixture$(TimeLimitedTests.scala:150)
	at app//infrastructure.TestBase.withFixture(TestBase.scala:9)
	at app//org.scalatest.FunSuiteLike.invokeWithFixture$1(FunSuiteLike.scala:184)
	at app//org.scalatest.FunSuiteLike.$anonfun$runTest$1(FunSuiteLike.scala:196)
	at app//org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
	at app//org.scalatest.FunSuiteLike.runTest(FunSuiteLike.scala:196)
	at app//org.scalatest.FunSuiteLike.runTest$(FunSuiteLike.scala:178)
	at app//org.scalatest.FunSuite.runTest(FunSuite.scala:1560)
	at app//org.scalatest.FunSuiteLike.$anonfun$runTests$1(FunSuiteLike.scala:229)
	at app//org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:393)
	at app//scala.collection.immutable.List.foreach(List.scala:333)
	at app//org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at app//org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:376)
	at app//org.scalatest.SuperEngine.runTestsImpl(Engine.scala:458)
	at app//org.scalatest.FunSuiteLike.runTests(FunSuiteLike.scala:229)
	at app//org.scalatest.FunSuiteLike.runTests$(FunSuiteLike.scala:228)
	at app//org.scalatest.FunSuite.runTests(FunSuite.scala:1560)
	at app//org.scalatest.Suite.run(Suite.scala:1124)
	at app//org.scalatest.Suite.run$(Suite.scala:1106)
	at app//org.scalatest.FunSuite.org$scalatest$FunSuiteLike$$super$run(FunSuite.scala:1560)
	at app//org.scalatest.FunSuiteLike.$anonfun$run$1(FunSuiteLike.scala:233)
	at app//org.scalatest.SuperEngine.runImpl(Engine.scala:518)
	at app//org.scalatest.FunSuiteLike.run(FunSuiteLike.scala:233)
	at app//org.scalatest.FunSuiteLike.run$(FunSuiteLike.scala:232)
	at app//infrastructure.TestBase.run(TestBase.scala:26)
	at app//org.scalatest.Suite.callExecuteOnSuite$1(Suite.scala:1187)
	at app//org.scalatest.Suite.$anonfun$runNestedSuites$1(Suite.scala:1234)
	at app//scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at app//org.scalatest.Suite.runNestedSuites(Suite.scala:1232)
	at app//org.scalatest.Suite.runNestedSuites$(Suite.scala:1166)
	at app//org.scalatest.Suites.runNestedSuites(Suites.scala:56)
	at app//org.scalatest.Suite.run(Suite.scala:1121)
	at app//org.scalatest.Suite.run$(Suite.scala:1106)
	at app//tetris.TetrisTestSuitesBase.runDirect(TetrisTestSuitesBase.scala:20)
	at app//tetris.TetrisTestSuitesBase.runWithScoreCounter(TetrisTestSuitesBase.scala:14)
	at app//tetris.TetrisTestSuite3_2.run(TetrisTestSuite3_2.scala:25)
	at app//org.scalatestplus.junit.JUnitRunner.run(JUnitRunner.scala:99)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.12/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.12/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
</failure>
  </testcase>
  <testcase name="testNoEscapeGameOver" classname="tetris.GameOverTests" time="0.002">
    <failure message="org.scalatest.exceptions.TestFailedException: didPass was false&#10;Hint: You cannot escape from a game over state by moving or rotating blocks.&#10;--------------------------------------------------------------------------------&#10;This is what went wrong:&#10;&#10;step=0, rand=1, actions=&lt;&gt;&#10;Want | Got ✓&#10;------------&#10;J... | J... &#10;JJJ. | JJJ. &#10;.... | .... &#10;&#10;step=1, rand=1, actions=&lt;Down&gt;&#10;Want | Got ✓&#10;------------&#10;.... | .... &#10;J... | J... &#10;JJJ. | JJJ. &#10;&#10;step=2, rand=1, actions=&lt;Down&gt;&#10;Want              | Got ✗&#10;-------------------------&#10;GameOverDisplay() | J... &#10;                  | JJJ. &#10;                  | JJJ. &#10;&#10;step=3, rand=1, actions=&lt;Right&gt;&#10;Want              | Got ✗&#10;-------------------------&#10;GameOverDisplay() | .J.. &#10;                  | JJJJ &#10;                  | JJJ. &#10;&#10;" type="org.scalatest.exceptions.TestFailedException">org.scalatest.exceptions.TestFailedException: didPass was false
Hint: You cannot escape from a game over state by moving or rotating blocks.
--------------------------------------------------------------------------------
This is what went wrong:

step=0, rand=1, actions=&lt;&gt;
Want | Got ✓
------------
J... | J... 
JJJ. | JJJ. 
.... | .... 

step=1, rand=1, actions=&lt;Down&gt;
Want | Got ✓
------------
.... | .... 
J... | J... 
JJJ. | JJJ. 

step=2, rand=1, actions=&lt;Down&gt;
Want              | Got ✗
-------------------------
GameOverDisplay() | J... 
                  | JJJ. 
                  | JJJ. 

step=3, rand=1, actions=&lt;Right&gt;
Want              | Got ✗
-------------------------
GameOverDisplay() | .J.. 
                  | JJJJ 
                  | JJJ. 


	at app//org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
	at app//org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:529)
	at app//org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
	at app//org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:503)
	at app//generic.GameTestSuite.checkGame(GameTestSuite.scala:298)
	at app//tetris.TetrisTestSuiteBase.checkGame(TetrisTestSuiteBase.scala:103)
	at app//tetris.GameOverTests.$anonfun$new$2(GameOverTests.scala:43)
	at app//scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at app//infrastructure.TestBase.$anonfun$test$1(TestBase.scala:36)
	at app//org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at app//org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at app//org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at app//org.scalatest.Transformer.apply(Transformer.scala:22)
	at app//org.scalatest.Transformer.apply(Transformer.scala:20)
	at app//org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:186)
	at app//org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at app//org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at app//infrastructure.TestBase.org$scalatest$concurrent$TimeLimitedTests$$super$withFixture(TestBase.scala:9)
	at app//org.scalatest.concurrent.TimeLimitedTests.$anonfun$withFixture$3(TimeLimitedTests.scala:154)
	at app//org.scalatest.enablers.Timed$$anon$1.timeoutAfter(Timed.scala:127)
	at app//org.scalatest.concurrent.TimeLimits.failAfterImpl(TimeLimits.scala:239)
	at app//org.scalatest.concurrent.TimeLimits.failAfterImpl$(TimeLimits.scala:233)
	at app//org.scalatest.concurrent.TimeLimits$.failAfterImpl(TimeLimits.scala:305)
	at app//org.scalatest.concurrent.TimeLimitedTests.withFixture(TimeLimitedTests.scala:153)
	at app//org.scalatest.concurrent.TimeLimitedTests.withFixture$(TimeLimitedTests.scala:150)
	at app//infrastructure.TestBase.withFixture(TestBase.scala:9)
	at app//org.scalatest.FunSuiteLike.invokeWithFixture$1(FunSuiteLike.scala:184)
	at app//org.scalatest.FunSuiteLike.$anonfun$runTest$1(FunSuiteLike.scala:196)
	at app//org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
	at app//org.scalatest.FunSuiteLike.runTest(FunSuiteLike.scala:196)
	at app//org.scalatest.FunSuiteLike.runTest$(FunSuiteLike.scala:178)
	at app//org.scalatest.FunSuite.runTest(FunSuite.scala:1560)
	at app//org.scalatest.FunSuiteLike.$anonfun$runTests$1(FunSuiteLike.scala:229)
	at app//org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:393)
	at app//scala.collection.immutable.List.foreach(List.scala:333)
	at app//org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at app//org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:376)
	at app//org.scalatest.SuperEngine.runTestsImpl(Engine.scala:458)
	at app//org.scalatest.FunSuiteLike.runTests(FunSuiteLike.scala:229)
	at app//org.scalatest.FunSuiteLike.runTests$(FunSuiteLike.scala:228)
	at app//org.scalatest.FunSuite.runTests(FunSuite.scala:1560)
	at app//org.scalatest.Suite.run(Suite.scala:1124)
	at app//org.scalatest.Suite.run$(Suite.scala:1106)
	at app//org.scalatest.FunSuite.org$scalatest$FunSuiteLike$$super$run(FunSuite.scala:1560)
	at app//org.scalatest.FunSuiteLike.$anonfun$run$1(FunSuiteLike.scala:233)
	at app//org.scalatest.SuperEngine.runImpl(Engine.scala:518)
	at app//org.scalatest.FunSuiteLike.run(FunSuiteLike.scala:233)
	at app//org.scalatest.FunSuiteLike.run$(FunSuiteLike.scala:232)
	at app//infrastructure.TestBase.run(TestBase.scala:26)
	at app//org.scalatest.Suite.callExecuteOnSuite$1(Suite.scala:1187)
	at app//org.scalatest.Suite.$anonfun$runNestedSuites$1(Suite.scala:1234)
	at app//scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at app//org.scalatest.Suite.runNestedSuites(Suite.scala:1232)
	at app//org.scalatest.Suite.runNestedSuites$(Suite.scala:1166)
	at app//org.scalatest.Suites.runNestedSuites(Suites.scala:56)
	at app//org.scalatest.Suite.run(Suite.scala:1121)
	at app//org.scalatest.Suite.run$(Suite.scala:1106)
	at app//tetris.TetrisTestSuitesBase.runDirect(TetrisTestSuitesBase.scala:20)
	at app//tetris.TetrisTestSuitesBase.runWithScoreCounter(TetrisTestSuitesBase.scala:14)
	at app//tetris.TetrisTestSuite3_2.run(TetrisTestSuite3_2.scala:25)
	at app//org.scalatestplus.junit.JUnitRunner.run(JUnitRunner.scala:99)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.12/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.12/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
</failure>
  </testcase>
  <testcase name="testNoEscapeGameOver2" classname="tetris.GameOverTests" time="0.011">
    <failure message="org.scalatest.exceptions.TestFailedException: didPass was false&#10;Hint: You cannot escape from a game over state by moving or rotating blocks.&#10;--------------------------------------------------------------------------------&#10;This is what went wrong:&#10;&#10;step=0, rand=1, actions=&lt;&gt;&#10;Want | Got ✓&#10;------------&#10;J... | J... &#10;JJJ. | JJJ. &#10;.... | .... &#10;.... | .... &#10;&#10;step=1, rand=1, actions=&lt;RotateRight&gt;&#10;Want | Got ✓&#10;------------&#10;.JJ. | .JJ. &#10;.J.. | .J.. &#10;.J.. | .J.. &#10;.... | .... &#10;&#10;step=2, rand=1, actions=&lt;Left&gt;&#10;Want | Got ✓&#10;------------&#10;JJ.. | JJ.. &#10;J... | J... &#10;J... | J... &#10;.... | .... &#10;&#10;step=3, rand=1, actions=&lt;Down&gt;&#10;Want | Got ✓&#10;------------&#10;.... | .... &#10;JJ.. | JJ.. &#10;J... | J... &#10;J... | J... &#10;&#10;step=4, rand=0, actions=&lt;Down&gt;&#10;Want              | Got ✗&#10;-------------------------&#10;GameOverDisplay() | .... &#10;                  | IIII &#10;                  | J... &#10;                  | J... &#10;&#10;step=5, rand=0, actions=&lt;RotateRight&gt;&#10;Want              | Got ✗&#10;-------------------------&#10;GameOverDisplay() | ..I. &#10;                  | JJI. &#10;                  | J.I. &#10;                  | J.I. &#10;&#10;" type="org.scalatest.exceptions.TestFailedException">org.scalatest.exceptions.TestFailedException: didPass was false
Hint: You cannot escape from a game over state by moving or rotating blocks.
--------------------------------------------------------------------------------
This is what went wrong:

step=0, rand=1, actions=&lt;&gt;
Want | Got ✓
------------
J... | J... 
JJJ. | JJJ. 
.... | .... 
.... | .... 

step=1, rand=1, actions=&lt;RotateRight&gt;
Want | Got ✓
------------
.JJ. | .JJ. 
.J.. | .J.. 
.J.. | .J.. 
.... | .... 

step=2, rand=1, actions=&lt;Left&gt;
Want | Got ✓
------------
JJ.. | JJ.. 
J... | J... 
J... | J... 
.... | .... 

step=3, rand=1, actions=&lt;Down&gt;
Want | Got ✓
------------
.... | .... 
JJ.. | JJ.. 
J... | J... 
J... | J... 

step=4, rand=0, actions=&lt;Down&gt;
Want              | Got ✗
-------------------------
GameOverDisplay() | .... 
                  | IIII 
                  | J... 
                  | J... 

step=5, rand=0, actions=&lt;RotateRight&gt;
Want              | Got ✗
-------------------------
GameOverDisplay() | ..I. 
                  | JJI. 
                  | J.I. 
                  | J.I. 


	at app//org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
	at app//org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:529)
	at app//org.scalatest.FunSuite.newAssertionFailedException(FunSuite.scala:1560)
	at app//org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:503)
	at app//generic.GameTestSuite.checkGame(GameTestSuite.scala:298)
	at app//tetris.TetrisTestSuiteBase.checkGame(TetrisTestSuiteBase.scala:103)
	at app//tetris.GameOverTests.$anonfun$new$3(GameOverTests.scala:69)
	at app//scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at app//infrastructure.TestBase.$anonfun$test$1(TestBase.scala:36)
	at app//org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at app//org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at app//org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at app//org.scalatest.Transformer.apply(Transformer.scala:22)
	at app//org.scalatest.Transformer.apply(Transformer.scala:20)
	at app//org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:186)
	at app//org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at app//org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at app//infrastructure.TestBase.org$scalatest$concurrent$TimeLimitedTests$$super$withFixture(TestBase.scala:9)
	at app//org.scalatest.concurrent.TimeLimitedTests.$anonfun$withFixture$3(TimeLimitedTests.scala:154)
	at app//org.scalatest.enablers.Timed$$anon$1.timeoutAfter(Timed.scala:127)
	at app//org.scalatest.concurrent.TimeLimits.failAfterImpl(TimeLimits.scala:239)
	at app//org.scalatest.concurrent.TimeLimits.failAfterImpl$(TimeLimits.scala:233)
	at app//org.scalatest.concurrent.TimeLimits$.failAfterImpl(TimeLimits.scala:305)
	at app//org.scalatest.concurrent.TimeLimitedTests.withFixture(TimeLimitedTests.scala:153)
	at app//org.scalatest.concurrent.TimeLimitedTests.withFixture$(TimeLimitedTests.scala:150)
	at app//infrastructure.TestBase.withFixture(TestBase.scala:9)
	at app//org.scalatest.FunSuiteLike.invokeWithFixture$1(FunSuiteLike.scala:184)
	at app//org.scalatest.FunSuiteLike.$anonfun$runTest$1(FunSuiteLike.scala:196)
	at app//org.scalatest.SuperEngine.runTestImpl(Engine.scala:286)
	at app//org.scalatest.FunSuiteLike.runTest(FunSuiteLike.scala:196)
	at app//org.scalatest.FunSuiteLike.runTest$(FunSuiteLike.scala:178)
	at app//org.scalatest.FunSuite.runTest(FunSuite.scala:1560)
	at app//org.scalatest.FunSuiteLike.$anonfun$runTests$1(FunSuiteLike.scala:229)
	at app//org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:393)
	at app//scala.collection.immutable.List.foreach(List.scala:333)
	at app//org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:381)
	at app//org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:376)
	at app//org.scalatest.SuperEngine.runTestsImpl(Engine.scala:458)
	at app//org.scalatest.FunSuiteLike.runTests(FunSuiteLike.scala:229)
	at app//org.scalatest.FunSuiteLike.runTests$(FunSuiteLike.scala:228)
	at app//org.scalatest.FunSuite.runTests(FunSuite.scala:1560)
	at app//org.scalatest.Suite.run(Suite.scala:1124)
	at app//org.scalatest.Suite.run$(Suite.scala:1106)
	at app//org.scalatest.FunSuite.org$scalatest$FunSuiteLike$$super$run(FunSuite.scala:1560)
	at app//org.scalatest.FunSuiteLike.$anonfun$run$1(FunSuiteLike.scala:233)
	at app//org.scalatest.SuperEngine.runImpl(Engine.scala:518)
	at app//org.scalatest.FunSuiteLike.run(FunSuiteLike.scala:233)
	at app//org.scalatest.FunSuiteLike.run$(FunSuiteLike.scala:232)
	at app//infrastructure.TestBase.run(TestBase.scala:26)
	at app//org.scalatest.Suite.callExecuteOnSuite$1(Suite.scala:1187)
	at app//org.scalatest.Suite.$anonfun$runNestedSuites$1(Suite.scala:1234)
	at app//scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
	at app//org.scalatest.Suite.runNestedSuites(Suite.scala:1232)
	at app//org.scalatest.Suite.runNestedSuites$(Suite.scala:1166)
	at app//org.scalatest.Suites.runNestedSuites(Suites.scala:56)
	at app//org.scalatest.Suite.run(Suite.scala:1121)
	at app//org.scalatest.Suite.run$(Suite.scala:1106)
	at app//tetris.TetrisTestSuitesBase.runDirect(TetrisTestSuitesBase.scala:20)
	at app//tetris.TetrisTestSuitesBase.runWithScoreCounter(TetrisTestSuitesBase.scala:14)
	at app//tetris.TetrisTestSuite3_2.run(TetrisTestSuite3_2.scala:25)
	at app//org.scalatestplus.junit.JUnitRunner.run(JUnitRunner.scala:99)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.12/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.12/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
</failure>
  </testcase>
  <system-out><![CDATA[]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
